pulsar/schema.go

```diff
@@ -0,0 +1,502 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package pulsar
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"reflect"
+	"unsafe"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/linkedin/goavro"
+)
+
+type SchemaType int
```

Comment:
Should we add a String() method on the enum then for the names we can call JSON.String(), etc... and it's useful for logging and or debugging

Reply:
Maybe we can use SchemaInfo.Name replace it, right?